plugins/lang/js/mldb_js.cc

```diff
@@ -79,6 +79,137 @@ registerMe()
+        } HANDLE_JS_EXCEPTIONS;
+    }
+
+    static v8::Handle<v8::Value>
+    when(const v8::Arguments & args)
+    {
+        try {
+            auto & val = getShared(args.This());
+            return JS::toJS(val.getEffectiveTimestamp());
+        } HANDLE_JS_EXCEPTIONS;
+    }
+
+    static v8::Handle<v8::Value>
+    at(const v8::Arguments & args)
+    {
+        try {
+            auto val = getShared(args.This());
+            Date newTs = JS::getArg<Date>(args, 0, "New effective timestamp");
+            val.setEffectiveTimestamp(newTs);
+            return JS::toJS(val);
+        } HANDLE_JS_EXCEPTIONS;
+    }
+
+    static v8::Handle<v8::Value>
+    columns(const v8::Arguments & args)
+    {
+        try {
+            auto & val = getShared(args.This());
+
+            v8::HandleScope scope;
+            v8::Local<v8::Object> obj= v8::Object::New();
+
+            auto onColumn = [&] (const PathElement & col,
+                                 const ExpressionValue & val)
+                {
+                    obj->Set(JS::toJS(col),
+                             JS::toJS(val));
+                    return true;
+                };
+
+            val.forEachColumn(onColumn);
+            return scope.Close(obj);
+        } HANDLE_JS_EXCEPTIONS;
+    }
+    
+
+#if 0
+    static v8::Handle<v8::Value>
+    getNamed(v8::Local<v8::String> property,
+             const v8::AccessorInfo& info)
+    {
+        try {
+            cerr << "get named" << endl;
+            auto val = getShared(info.This());
+            return v8::Undefined();
+        } HANDLE_JS_EXCEPTIONS;
+    }
+
+    static v8::Handle<v8::Integer>
+    queryNamed(v8::Local<v8::String> property,
+               const v8::AccessorInfo& info)
+    {
+        cerr << "query named" << endl;
+        auto val = getShared(info.This());
+        abort();
+    }
+
+    static v8::Handle<v8::Array>
+    enumerateNamed(const v8::AccessorInfo& info)
+    {
+        cerr << "enumerate named" << endl;
+        auto val = getShared(info.This());
+        abort();
+    }
+#endif
+};
+
+v8::Handle<v8::Object>
+ExpressionValueJS::
+create(ExpressionValue value, JsThreadContext * context)
+{
+    auto obj = context->ExpressionValue->GetFunction()->NewInstance();
+    auto * wrapped = new ExpressionValueJS();
+    wrapped->val = std::move(value);
+    wrapped->wrap(obj, context);
+    return obj;
+}
+
+ExpressionValue &
+ExpressionValueJS::
+getShared(const v8::Handle<v8::Object> & val)
+{
+    return reinterpret_cast<ExpressionValueJS *>
+        (v8::Handle<v8::External>::Cast
+         (val->GetInternalField(0))->Value())->val;
+}
+
+v8::Local<v8::FunctionTemplate>
+ExpressionValueJS::
+registerMe()
+{
+    using namespace v8;
+
+    HandleScope scope;
+
+    auto fntmpl = CreateFunctionTemplate("ExpressionValue");
+    auto objtmpl = fntmpl->PrototypeTemplate();
+
+    objtmpl->Set(String::New("toJs"), FunctionTemplate::New(Methods::toJs));
```

Comment:
Might be more forgiving for the user to register also the function `tojs` (without a capital j) since this is the only function name containing a capital.